src/audits/performance/user-timings.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Audit = require('../audit');
+const TimeLineModel = require('../../lib/traces/devtools-timeline-model');
+
+const FAILURE_MESSAGE = 'Trace data not found.';
+
+class UserTimings extends Audit {
+  /**
+   * @override
+   */
+  static get category() {
+    return 'Performance';
+  }
+
+  /**
+   * @override
+   */
+  static get name() {
+    return 'user-timings';
+  }
+
+  /**
+   * @override
+   */
+  static get description() {
+    return 'blink.user_timing events';
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    if (!artifacts.traceContents || !Array.isArray(artifacts.traceContents)) {
+      throw new Error(FAILURE_MESSAGE);
+    }
+
+    // Fetch blink.user_timing events from the tracing data
+    const model = new TimeLineModel(artifacts.traceContents);
+    const tModel = model.timelineModel();
+    const group = [...tModel.mainThreadAsyncEvents().keys()].find(
+      group => group.title === 'User Timing'
+    );
+    let userTimings = tModel.mainThreadAsyncEvents().get(group);
+
+    // Reduce events to record only useful information
+    if (typeof userTimings !== 'undefined') {
+      userTimings = userTimings.map(ut => {
+        return {
+          name: ut.name,
+          startTime: ut.startTime,
+          endTime: ut.endTime,
+          duration: ut.duration,
+          args: ut.args
+        };
+      });
+    }
+
+    return UserTimings.generateAuditResult({
+      value: true,
+      extendedInfo: {
+        /* Pass empty array rather than undefined */
+        value: userTimings || []
```

Comment:
We need some way of handling this in the report. Are you doing that as part of this PR, or is someone else taking it?